group_theory/perm.lean

```diff
@@ -261,12 +330,42 @@ def sign [fintype α] (f : perm α) := sign_aux3 f mem_univ
 instance sign.is_group_hom [fintype α] : is_group_hom (@sign α _ _) :=
 ⟨λ f g, (sign_aux3_mul_and_swap f g _ mem_univ).1⟩
 
+@[simp] lemma sign_mul [fintype α] (f g : perm α) : sign (f * g) = sign f * sign g :=
+is_group_hom.mul sign _ _
+
+@[simp] lemma sign_one [fintype α] : (sign (1 : perm α)) = 1 :=
+is_group_hom.one sign
+
+@[simp] lemma sign_inv [fintype α] (f : perm α) : sign f⁻¹ = sign f :=
+by rw [is_group_hom.inv sign, int.units_inv_eq_self]; apply_instance
+
 lemma sign_swap [fintype α] {x y : α} (h : x ≠ y) : sign (swap x y) = -1 :=
 (sign_aux3_mul_and_swap 1 1 _ mem_univ).2 x y h
 
 lemma sign_eq_of_is_swap [fintype α] {f : perm α} (h : is_swap f) : sign f = -1 :=
 let ⟨x, y, hxy⟩ := h in hxy.2.symm ▸ sign_swap hxy.1
 
+lemma sign_symm_trans_trans [fintype α] [decidable_eq β] [fintype β] (f : perm α)
+  (e : α ≃ β) : sign ((e.symm.trans f).trans e) = sign f :=
+show sign_aux3 ((e.symm.trans f).trans e) (show ∀ x : β, x ∈ univ.1, from mem_univ) =
+  sign_aux3 f (show ∀ x : α, x ∈ univ.1, from mem_univ),
+from quotient.induction_on₂ (@univ β _).1 (@univ α _).1
+  (λ l₁ l₂ h₁ h₂, show sign_aux2 _ _ = sign_aux2 _ _,
+    from let n := trunc.out (equiv_fin β) in
+    by rw [← sign_aux_eq_sign_aux2 _ _ n (λ _ _, h₁ _),
+        ← sign_aux_eq_sign_aux2 _ _ (e.trans n) (λ _ _, h₂ _)];
+      exact congr_arg sign_aux (equiv.ext _ _ (λ x, by simp)))
+(show ∀ x : β, x ∈ univ.1, from mem_univ)
+(show ∀ x : α, x ∈ univ.1, from mem_univ)
+
+lemma sign_eq_pow_length_swap_factors [fintype α] {f : perm α} {l : list (perm α)}
+  (h : l.prod = f) (hl : ∀ g ∈ l, is_swap g) : sign f = -1 ^ l.length :=
```

Comment:
is it necessary to have `h` and `f` why not state it as `sign l.prod = -1 ^ l.length`?